native/spark-expr/src/cast.rs

```diff
@@ -724,26 +732,31 @@ impl Cast {
             .downcast_ref::<GenericStringArray<i32>>()
             .expect("Expected a string array");
 
-        let cast_array: ArrayRef = match to_type {
-            DataType::Date32 => {
-                let len = string_array.len();
-                let mut cast_array = PrimitiveArray::<Date32Type>::builder(len);
-                for i in 0..len {
-                    if !string_array.is_null(i) {
-                        match date_parser(string_array.value(i), eval_mode) {
-                            Ok(Some(cast_value)) => cast_array.append_value(cast_value),
-                            Ok(None) => cast_array.append_null(),
-                            Err(e) => return Err(e),
-                        }
-                    } else {
-                        cast_array.append_null()
-                    }
+        if to_type != &DataType::Date32 {
```

Comment:
I don't see a logic change here. Is this just refactoring?

Reply:
yes, no logic change here.